client/src/app/services/user.service.ts

```diff
@@ -0,0 +1,106 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+import { Token } from '../interfaces/token';
+import { User, UserSettingKeyValue } from '../models/user.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+
+/**
+ * User service to store and retrieve user.
+ */
+export class UserService {
+  localStorageUserKey: string = 'user';
+  currentUser: User | null;
+  userWatch: BehaviorSubject<User | null> = new BehaviorSubject<User | null>(
+    null
+  );
+
+  /**
+   * Constructor.
+   */
+  constructor() {
+    this.loadFromLocalStorage();
+  }
+
+  /**
+   * Get user from local storage.
+   */
+  loadFromLocalStorage() {
+    if (localStorage.getItem(this.localStorageUserKey)) {
+      this.user = User.fromJSON(localStorage.getItem(this.localStorageUserKey));
+    }
+  }
+
+  /**
+   * Get logged in user.
+   *
+   * @returns {User|null}
+   */
+  get user(): User | null {
+    return this.currentUser;
+  }
+
+  /**
+   * Set user.
+   *
+   * @param {User|null} user
+   */
+  set user(user: User | null) {
+    this.currentUser = user;
+
+    // Write to LocalStorage
+    localStorage.setItem(
```

Comment:
I'd consider checking if the the user is null and then writing the user JSON to local storage only in case you have a non-null user, otherwise removing the entry. Then you can check for the existence of a user entry in local storage via "has()".